app/Http/Controllers/AlbumsController.php

```diff
@@ -70,78 +72,29 @@ public function getPositionData()
 
 		// Initialize return var
 		$return = array();
-		$return['albums'] = array();
 
-		// Get photos
-		// Get album information
-		$UserId = $this->sessionFunctions->id();
-		$full_photo = Configs::get_value('full_photo', '1') == '1';
-
-		$toplevel = $this->albumFunctions->getToplevelAlbums();
-		if ($toplevel === null) {
-			return Response::error('I could not find you.');
-		}
-
-		// All normal albums
-		$toplevel_albums = $this->albumFunctions->prepare_albums($toplevel['albums']);
-		// Add albums shared with users
-		$toplevel_albums = array_merge($toplevel_albums, $this->albumFunctions->prepare_albums($toplevel['shared_albums']));
-
-		$toplevel_album_ids = array();
-		$album_list = array();
-
-		foreach ($toplevel_albums as $album_iterator) {
-			$toplevel_album_ids[] = $album_iterator['id'];
-
-			$album_tmp = Album::with('children')->find($album_iterator['id']);
-			$full_photo = $album_tmp->full_photo_visible();
+		$albumIDs = $this->albumFunctions->getPublicAlbums();
+
+		$query = Photo::with('album')->where(
+				function (Builder $query) use ($albumIDs) {
+					$query->whereIn('album_id', $albumIDs);
```

Comment:
This line does nothing.

Reply:
Erm, sure it does: it adds all the accessible albums to the query.

Let's recap what we are trying to do here so that there are no misunderstandings: we want a query that is going to go over all the photos accessible to the current user by simple browsing from the top level (i.e., hidden and password-protected albums excluded). `AlbumFunctions::getPublicAlbums()` already did the heavy lifting, and its result is in `$albumIDs`. We "just" need to add the unsorted photos. Which is tricky, since they don't belong to an album, and in multi-user case, are mixed together so also need to be filtered by `owner id`. That's what the code below this line does, and I don't think that the desired effect can be achieved with a static query, though I'll be happy to be proven wrong. At most, the line above can probably be moved outside the function, but will that change anything?

So what is your concern here exactly?

Reply:
nope, it should be `$query = $query->whereIn('album_id', ...)` or I am wrong. :)

Reply:
I totally missed that! Now I'm beginning to wonder how things work at all in the `SearchController` because there's the same bug there...

Reply:
No, wait... Look at https://github.com/LycheeOrg/Lychee-Laravel/blob/master/app/Http/Controllers/SearchController.php#L213-L221. It's the same "erroneous" syntax yet that one definitely has an effect?!